### PR TITLE
Get rid of kruft LogObject copy constructor.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

File: proxy/logging/LogObject.h [hunk 1]
```diff
@@ -97,7 +97,6 @@ class LogObject : public RefCountObj
             const char *header, Log::RollingEnabledValues rolling_enabled, int flush_threads, int rolling_interval_sec = 0,
             int rolling_offset_hr = 0, int rolling_size_mb = 0, bool auto_created = false, int rolling_max_count = 0,
             int rolling_min_count = 0, bool reopen_after_rolling = false, int pipe_buffer_size = 0);
-  LogObject(LogObject &);
```
Comment (bneradt) [inline proxy/logging/LogObject.h:100]: We should probably delete this and the copy assignment constructors to make sure they don't accidentally get used in the future.

> LogObject(LogObject const&) = delete;
> LogObject& operator=(LogObject const&) = delete;

**Review by bneradt — CHANGES_REQUESTED** [review]
Good idea. It's best to delete this rather than maintain something not used.

File: proxy/logging/LogObject.h (hunk 1 above)
Comment (ywkaras) [reply proxy/logging/LogObject.h:100]: These are already present:  https://github.com/apache/trafficserver/blob/e6291181b4eb6bc98b57fa1cd86ec07fd2d59928/proxy/logging/LogObject.h#L300
Although they are private, which would make for a confusing error message I think.  So the defined copy constructor is especially krufty, it only applies to non-const source objects.

Comment (bneradt) [reply proxy/logging/LogObject.h:100]: Ah, ok. Thanks!

**Review by bneradt — APPROVED** [review]

**Review by bryancall — APPROVED** [review]

**Review by vmamidi — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**